File: sync_shared_footer.py
from pathlib import Path
import re

ROOT = Path(__file__).resolve().parent
PARTIAL = (ROOT / "partials" / "footer.html").read_text(encoding="utf-8").strip()
STYLE_VERSION = "20260830_footer_canonical_v2"
FOOTER_PATTERN = re.compile(
    r'<footer\s+class=["\'](?:main-footer|footer)["\'][^>]*>[\s\S]*?</footer>',
    re.IGNORECASE,
)
EXCLUDED = {"admin.html", "login.html", "petugas.html", "lpg-agen.html", "profil-petugas.html"}

def sync_all(target_root: Path = ROOT) -> int:
    updated = 0
    target_root = Path(target_root).resolve()
    for path in target_root.rglob("*.html"):
        if path.name in EXCLUDED or ".git" in path.parts:
            continue
        source = path.read_text(encoding="utf-8")
        result, count = FOOTER_PATTERN.subn(PARTIAL, source)
        if count:
            result = re.sub(
                r'(css/style\.css)(?:\?[^"\']*)?',
                rf'\1?v={STYLE_VERSION}',
                result,
            )
        if count and result != source:
            path.write_text(result, encoding="utf-8", newline="\n")
            updated += 1
    return updated


if __name__ == "__main__":
    print(f"Footer kanonis diterapkan ke {sync_all()} halaman.")

File: scripts/sync_shared_footer.mjs
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const SOURCE_ROOT = path.resolve(path.dirname(fileURLToPath(import.meta.url)), "..");
const ROOT = process.argv[2] ? path.resolve(process.argv[2]) : SOURCE_ROOT;
const partial = fs.readFileSync(path.join(SOURCE_ROOT, "partials", "footer.html"), "utf8").trim();
const excluded = new Set(["admin.html", "login.html", "petugas.html", "lpg-agen.html", "profil-petugas.html"]);
function walk(dir) {
  return fs.readdirSync(dir, {withFileTypes:true}).flatMap(entry => {
    if ([".git", ".firebase", "dist"].includes(entry.name)) return [];
    const item = path.join(dir, entry.name);
    return entry.isDirectory() ? walk(item) : entry.name.endsWith(".html") ? [item] : [];
  });
}
let count = 0;
for (const file of walk(ROOT)) {
  if (excluded.has(path.basename(file))) continue;
  const source = fs.readFileSync(file, "utf8");
  if (!/<footer\s+class=["'](?:main-footer|footer)["'][^>]*>[\s\S]*?<\/footer>/i.test(source)) continue;
  const updated = source
    .replace(/<footer\s+class=["'](?:main-footer|footer)["'][^>]*>[\s\S]*?<\/footer>/i, partial)
    .replace(/css\/style\.css(?:\?[^"']*)?/g, "css/style.css?v=20260902_directory_nav_v1");
  if (updated !== source) { fs.writeFileSync(file, updated.replace(/\r\n/g,"\n")); count++; }
}
console.log(`Footer tersinkron: ${count} halaman`);
